### ISSUE TITLE
Code Coverage Report from @ui5/webcomponents-tools

### ISSUE THREAD

**Issue by codefactor** [SAP/ui5-webcomponents#1767]
**Is your feature request related to a problem? Please describe.**
We need a way to report on code coverage of our ui5 web component project, e.g. 80% code coverage is a requirement. The unit tests, though, do not expose this information that I can find. Is this something that is possible?

**Describe the solution you'd like**
A new script like `wc-dev coverage` that can produce a coverage report like jest or karma has.

**Describe alternatives you've considered**
Using karma instead of @ui5/webcomponents-tools

**Comment by codefactor** [comment]
@vladitasev ,
Just to update you, we utilized https://www.npmjs.com/package/@open-wc/testing-karma
I got it to work now and it works well with the build too, so I was able to merge it. Using the files that the build generates into the `/coverage` directory we can get a detailed breakdown of the coverage, which at this point is very low. I then proceeded and created unit test cases for each of our components. For now we will work primarily with the unit test cases in karma, and maybe later we can set up the integration tests.

There are just a few issues though:
1. Running the `npm run test:watch` requires me to run `npm start` in the background so that when I change files the `npm start` will auto-copy the files into the `/dist` folder. This works for now, but it would be nice to have a single command to run that did not also listen to port 8081 unnecessarily
2. There is some issue with the karma configuration which has trouble to import Assets, it gives some dynamic import issue - and without the Assets a bunch of error logs happen - so I fake out the assets a bit to work around the log error messages

Maybe you can try out the project here:
https://github.wdf.sap.corp/xweb/common-webcomponents

The latest code is now merged to master, so you can clone this repo and try it out. I think this might be useful to integrate this into the `@ui5/webcomponents-tools/components-package/nps.js` to support unit test cases and code coverage into any ui5/webcomponent based library

**Comment by vladitasev** [comment]
@codefactor Would the linked change give you enough flexibility for the purpose of your unit tests setup?
As listed in the PR description, there would be three new test-related commands:
-  `yarn serve` (just the server)
- `yarn test --no-server`  (just the tests)
- `yarn watch test` (just the watch)

**Comment by codefactor** [comment]
@vladitasev ,

The `wc-dev watch test` command is working good for me. I now have a package.json script that looks like this:

```
.....
  "scripts": {
......
    "test:build-watch": "wc-dev watch test",
    "test:karma-watch": "karma start --auto-watch=true --single-run=false",
    "test:watch": "concurrently --kill-others \"npm run test:build-watch\" \"npm run test:karma-watch\""
  }
....
```
Then when I run
```
npm run test:watch
```
This will in parallel execute the `wc-dev watch test` and the `karma start --auto-watch=true --single-run=false` then achieving the desired effect which is to re-run the karma unit tests any time a code change is made.

**Comment by codefactor** [comment]
@vladitasev ,

Sorry, I think I was already running a different watch process in the background so that's why it was working - when I stopped other processes it doesn't seem to work. I will set up a test case for you to try out soon.

**Comment by codefactor** [comment]
@vladitasev ,

The issue was that my script was doing `wc-dev watch test` and it was supposed to do `wc-dev watch src` - now it is working as expected.

**Comment by ilhan007** [comment]
Hello @codefactor @vladitasev reading the last comment, it seems there is a working solution adopted by @codefactor and the issue could be closed.
